yasa/evaluation.py

```diff
@@ -0,0 +1,649 @@
+    TST      True      True           True
+
+    Access more detailed statistical output of each test.
+
+    >>> sse.normality
+                  W      pval  normal
+    sstat
+    %N1    0.973407  0.824551    True
+    %N2    0.960684  0.557595    True
+    %N3    0.958591  0.516092    True
+    %REM   0.901733  0.044447   False
+    SE     0.926732  0.133580    True
+    SOL    0.774786  0.000372   False
+    TST    0.926733  0.133584    True
+    WASO   0.924288  0.119843    True
+
+    >>> sse.homoscedasticity.head(2)
+                  W      pval  equal_var
+    sstat
+    %N1    0.684833  0.508274       True
+    %N2    0.080359  0.922890       True
+
+    >>> sse.proportional_bias.round(3).head(2)
+            coef     se      T   pval     r2  adj_r2  CI[2.5%]  CI[97.5%]  unbiased
+    sstat
+    %N1   -0.487  0.314 -1.551  0.138  0.118   0.069    -1.146      0.172      True
+    %N2   -0.107  0.262 -0.409  0.688  0.009  -0.046    -0.658      0.444      True
+
+    .. plot::
+
+        >>> import matplotlib.pyplot as plt
+        >>> ax = sse.plot_discrepancies_heatmap()
+        >>> ax.set_title("Sleep statistic discrepancies")
+        >>> plt.tight_layout()
+
+    .. plot::
+
+        >>> sse.plot_blandaltman()
+    """
+    def __init__(self, data, reference, test, subject, statistic):
+        assert isinstance(data, pd.DataFrame), "`data` must be a pandas DataFrame"
+        for col in [reference, test, subject, statistic]:
+            assert isinstance(col, str) and col in data, f"`{col}` must be a string and a column in `data`"
+        assert data[subject].nunique() > 1, "`data` must include more than one subject"
+        data = data.copy()
+
+        # Get measurement difference between reference and test devices
+        data["difference"] = data[test].sub(data[reference])
+
+        # Check for sleep statistics that have no differences between measurement devices.
+        # This is most likely to occur with TIB but is possible with any, and will break some functions.
+        stats_nodiff = data.groupby(statistic)["difference"].any().loc[lambda x: ~x].index
+        for s in stats_nodiff:
+            data = data.query(f"{statistic} != '{s}'")
+            logger.warning(f"All {s} differences are zero, removing from evaluation.")
+            ## Q: Should this be logged as just info?
+
+        # Get list of all statistics to be evaluated
+        self.all_sleepstats = data[statistic].unique()
+
+        # Save attributes
+        self.data = data
+        self.reference = reference
+        self.test = test
+        self.subject = subject
+        self.statistic = statistic
+
+        # Run tests
+        self.test_normality(method="shapiro", alpha=0.05)
+        self.test_proportional_bias(alpha=0.05)
+        self.test_homoscedasticity(method="levene", alpha=0.05)
+
+    def test_normality(self, **kwargs):
+        """Test reference data for normality at each sleep statistic.
```

Comment:
Shouldn't it be the test for normality of the differences test - ref?

Reply:
Yikes good catch.